test/functional/specs/C14409.js

```diff
@@ -0,0 +1,96 @@
+import { t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../helpers/networkLogger";
+import getResponseBody from "../helpers/networkLogger/getResponseBody";
+import fixtureFactory from "../helpers/fixtureFactory";
+import cookies from "../helpers/cookies";
+import alloyEvent from "../helpers/alloyEvent";
+
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title: "C14409 - Consenting to no purposes should be persisted.",
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C14409",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const setConsentToNone = ClientFunction(() => {
+  return window.alloy("setConsent", { general: "out" });
+});
+
+test("C14409 - Consenting to no purposes should be persisted.", async () => {
+  const imsOrgId = "53A16ACB5CC1D3760A495C99@AdobeOrg";
+  await cookies.clear();
+
+  const configure = await alloyEvent("configure", {
+    configId: "9999999",
+    orgId: imsOrgId,
+    debugEnabled: true,
+    idMigrationEnabled: false
+  });
+
+  await configure.promise;
+
+  // send alloy event
+  const event1 = await alloyEvent({
+    viewStart: true
+  });
+
+  // apply user consent
+  await setConsentToNone();
+
+  await event1.promise;
+
+  const cookieName = `kndctr_${imsOrgId.replace(
+    /[@]+?/,
+    "_"
+  )}_consent`.toString();
+
+  const consentCheck = await cookies.get(cookieName);
+
+  await t.expect(consentCheck).eql("general=out");
+
+  // test konductor response
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
+  await t
+    .expect(networkLogger.edgeEndpointLogs.requests[0].response.statusCode)
+    .eql(200);
+
+  const body = getResponseBody(networkLogger.edgeEndpointLogs.requests[0]);
+  const bodyObject = JSON.parse(body);
+
+  // read state:store handles from response (i.e. 'set a cookie')
+  const cookiesToSet = {};
+
+  await t.expect("handle" in bodyObject).ok();
+  await t.expect(bodyObject.handle.length).gt(0);
+
+  await bodyObject.handle.map(handle => {
+    if (handle.type !== "state:store") return handle;
+    handle.payload.map(payload => {
+      cookiesToSet[payload.key] = payload;
+      return payload;
+    });
+    return handle;
+  });
+
+  const errorMessage = await t.eval(() =>
+    window
+      .alloy("event", { data: { a: 1 } })
+      .then(() => undefined, e => e.message)
+  );
+
+  // expect that konductor cookie handle matches cookie name
+  await t.expect(cookieName in cookiesToSet).ok();
+
+  await t.expect(errorMessage).ok("Expected the event command to be rejected");
+  await t.expect(errorMessage).contains("The user declined consent.");
+
+  // expect that the cookie max age is greater than 0
+  await t.expect("maxAge" in cookiesToSet[cookieName]).ok();
+  await t.expect(cookiesToSet[cookieName].maxAge).gt(0);
+});
```

Comment:
I think there are a couple things missing here:

1. We should be testing that `maxAge` is set to `15552000` (I didn't have this number when I wrote the TestRails tests). Ideally, we would be able to check the maxAge of the actual cookie (rather than what's in the edge response), because we'd like to test that Alloy actually writes the cookie using whatever maxAge was in the response. I don't know if this is actually possible with TestRail though. It's not normally possible with JavaScript in a regular browser. If it's not possible in TestRail, then checking the the maxAge in the response like you're doing here is probably the best we can do. Let's check that the value is actually `15552000` though.
1. We should be testing that the cookie `domain` is set to the apex (top-level) domain. Again, TestCafe may not let us do this by inspecting the cookie directly. You can't using JavaScript in a regular browser. However, we can still check that it's set using the apex domain if we navigate to a different subdomain before executing the event command. If the consent cookie is set on the apex domain, then the cookie will be accessible on the second subdomain and respected by Alloy. If the cookie were _not_ set on the apex domain, then the cookie would _not_ be accessible on the second domain and therefore not respected by Alloy. I'm not sure what the status is of being able to navigate to different subdomains within e2e tests, so I've posted a message in Slack.

Reply:
Saw your slack message. It looks like there aren't subdomains on alloyio.com for testing yet. Do you know if there's a way to simulate it locally?

Reply:
You could probably mimic it on your machine by manually editing your `/etc/hosts` file, but I think @shammowla is working on something where you wouldn't have to do that. I wouldn't bother with it yet or let it hold up your PR. In your PR I would just reload the current page, then after we get subdomain support for our testing purposes, we can come back and update this test to navigate to a page on the different subdomain rather than just reloading the current page.

Reply:
Sounds good! I'll go with reload for now and update the `maxAge` assertion. Thanks!